include/aws/crt/mqtt/Mqtt5Packets.h

```diff
@@ -448,10 +448,7 @@ namespace Aws
             class AWS_CRT_CPP_API NegotiatedSettings
             {
               public:
-                NegotiatedSettings(
-                    const aws_mqtt5_negotiated_settings &negotiated_settings,
-
-                    Allocator *allocator = ApiAllocator()) noexcept;
+                NegotiatedSettings(const aws_mqtt5_negotiated_settings &negotiated_settings) noexcept;
 
```

Comment:
Don't change this. The pattern is that everything takes a custom allocator. This datastructure just happens to not need it at the moment

instead, go into the .cpp file and mark it unused via `(void)allocator;`